### PR TITLE
Feature/cfa single partners cms

### LINKED ISSUE #582: @/codeforafrica Single Partner Page
Move single partners page to payload.

Ideally, this should be implemented in PR #581 but broke it down to make the PR reviewable.

[Current implementation](https://cfa.dev.codeforafrica.org/about/partners/afd)

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

This PR
- Migrates Single partners page to CMS
- Implements pagify for single page view

Fixes #582 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/components/ConnectBar/ConnectBar.js [hunk 1]
```diff
@@ -6,18 +6,14 @@ import SocialMediaButton from "@/codeforafrica/components/SocialMediaButton";
 const ConnectBar = React.forwardRef(function ConnectBar(props, ref) {
   const { sx, title, links } = props;
 
-  if (!links || !Object.entries(links)?.length) {
+  if (!links || !links?.length) {
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/ConnectBar/ConnectBar.js:9]:
```suggestion
  if (!links?.length) {
```

Isn't this enough?

File: apps/codeforafrica/src/components/SinglePartner/SinglePartner.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { Section } from "@commons-ui/core";
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:1]: Why a new component? What where we using before?

File: apps/codeforafrica/src/lib/data/common/index.js [hunk 3]
```diff
@@ -56,7 +58,10 @@ export async function getPageProps(api, context) {
   const settings = await api.findGlobal("settings");
   const navbar = getNavBar(settings);
   const footer = getFooter(settings);
-  const blocks = await blockify(page.blocks);
+  let blocks = await blockify(page.blocks);
+  if (params?.slugs?.length > 2) {
+    blocks = await pagify(slug, api, context);
+  }
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/common/index.js:64]: Been a while, so pagify also just returns blocks? I'd have though pagify creates a "stand-in" page that we can then process blocks from.

File: apps/codeforafrica/src/payload/collections/Partners.js [hunk 4]
```diff
@@ -60,9 +60,22 @@ const Partners = {
...
+            beforeValidate: [nestCollectionUnderPage("partners")],
+          },
+        },
+      ],
+    },
   ],
-  hooks: {
-    afterRead: [nestCollectionUnderPage("partners")],
-  },
 };
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/collections/Partners.js:80]: Why are we switching from collection hook to field hook? Isn't this breaking other parts that are using nestCollectionUnderPage as collection hook?

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/codeforafrica/src/components/SinglePartner/SinglePartner.js (hunk 2 above)
Comment (koechkevin) [reply apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:1]: @kilemensi , before this component was implemented in a [page](https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/src/pages/about/%5Bunit%5D/index.page.js)

File: apps/codeforafrica/src/payload/collections/Partners.js (hunk 4 above)
Comment (koechkevin) [reply apps/codeforafrica/src/payload/collections/Partners.js:80]: I still can't figure out why the collection hook crashes the app after sometime.

**Review by kelvinkipruto — APPROVED** [review]
LGTM

**Comment by kelvinkipruto** [comment]
@koechkevin Update the Partners block preview image

**Comment by koechkevin** [comment]
> @koechkevin Update the Partners block preview image

https://codeforafrica-ui-pr-585.dev.codeforafrica.org/about

File: apps/codeforafrica/src/components/SinglePartner/SinglePartner.js (hunk 2 above)
Comment (kilemensi) [reply apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:1]: Not sure I follow @koechkevin. If this was build in a page, it means it's just a collection of other components, no?

Comment (kilemensi) [reply apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:1]: @koechkevin ?

Comment (koechkevin) [reply apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:1]: Yes, it is a collection of other components but there's some extra styles added to the components so I thought a standalone page would work best @kilemensi

**Comment by kelvinkipruto** [comment]
> > @koechkevin Update the Partners block preview image
> 
> https://codeforafrica-ui-pr-585.dev.codeforafrica.org/about

@Koech I meant the image preview in the CMS similar to https://github.com/CodeForAfrica/ui/blob/35caf0d5b6673873ede01130240aec2497469df4/apps/codeforafrica/src/payload/blocks/CustomPageHeader.js#L5

File: apps/codeforafrica/src/payload/collections/Partners.js (hunk 4 above)
Comment (koechkevin) [reply apps/codeforafrica/src/payload/collections/Partners.js]: Yes @kilemensi

File: apps/codeforafrica/src/components/AboutChildPageHeader/AboutChildPageHeader.js [hunk 5]
```diff
@@ -69,7 +69,7 @@ const AboutChildPageHeader = React.forwardRef(
               <MemberFigureRoot
                 {...FigureProps}
                 sx={{
-                  background: `url(${image.src})`,
+                  background: `url(${image.url})`,
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AboutChildPageHeader/AboutChildPageHeader.js:72]: Can we standardise on image.src?

File: apps/codeforafrica/src/components/SinglePartner/SinglePartner.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            "& > p": {
+              mb: 5,
+            },
+          }}
+          elements={description}
+        />
+
+        <ConnectBar title="Connect" links={connect} />
+      </Section>
+      {relatedProjects.length ? (
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/SinglePartner/SinglePartner.js:58]:
```suggestion
      {relatedProjects?.length ? (
```

What if there aren't any related projects?

File: apps/codeforafrica/src/lib/data/blockify/index.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ const propsifyBlockBySlug = {
 
 async function blockify(blocks) {
   const promises = blocks?.map(async (block) => {
-    const slug = block.blockType;
+    const slug = block.blockType ?? block.slug ?? null;
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/blockify/index.js:9]: Is there a `block` without `blockType`?

File: apps/codeforafrica/src/lib/data/pagify/index.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import partners from "./partners";
+
+const COLLECTION_BY_SLUG = {
+  partners,
+};
+
+async function pagify(parentPage, api, context) {
+  const { slug: collection } = parentPage;
+  const pageProps = COLLECTION_BY_SLUG[collection];
```
Comment (kilemensi) [inline apps/codeforafrica/src/lib/data/pagify/index.js:9]: What if pageProps is undefined?

File: apps/codeforafrica/src/payload/blocks/OurPartners.js [hunk 9]
```diff
@@ -1,5 +1,7 @@
 const Partners = {
   slug: "our-partners",
+  imageURL: "/images/cms/blocks/partners.png",
+  imageAltText: "Our Partners List",
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/blocks/OurPartners.js:4]:
```suggestion
  imageAltText: "Our Partners",
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

There is our partners on homepage and our partners on about page. How are we handling these?

**Comment by koechkevin** [comment]
> 👍🏽
> 
> There is our partners on homepage and our partners on about page. How are we handling these?

@kilemensi My thinking is, users can add a block on each page and select from a list of partners(collection) to show.

File: apps/codeforafrica/src/components/AboutChildPageHeader/AboutChildPageHeader.js (hunk 5 above)
Comment (koechkevin) [reply apps/codeforafrica/src/components/AboutChildPageHeader/AboutChildPageHeader.js:72]: Didn't quite get what you meant here @kilemensi. Does your proposal imply adding a `src` field to Media?

**Comment by kilemensi** [comment]
> > 👍🏽
> > There is our partners on homepage and our partners on about page. How are we handling these?
> 
> @kilemensi My thinking is, users can add a block on each page and select from a list of partners(collection) to show.

We have this [function](https://github.com/CodeForAfrica/ui/blob/develop/apps/codeforafrica/src/lib/data/utils/index.js) for that @koechkevin

File: apps/codeforafrica/src/components/Partner/Partner.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import { Box } from "@mui/material";
+import React from "react";
+
+import AboutChildPageHeader from "@/codeforafrica/components/AboutChildPageHeader";
+import ConnectBar from "@/codeforafrica/components/ConnectBar";
+import RelatedProjects from "@/codeforafrica/components/RelatedProjects";
+import RichText from "@/codeforafrica/components/RichText";
+import SectionDivider from "@/codeforafrica/components/SectionDivider";
+
+const SinglePartner = React.forwardRef(function SinglePartner(
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/Partner/Partner.js:11]:
```suggestion
const Partner = React.forwardRef(function Partner(
```

File: apps/codeforafrica/src/components/Partner/Partner.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            title={relatedProjectsTitle}
+            projects={relatedProjects}
+          />
+        </>
+      ) : null}
+    </Box>
+  );
+});
+
+export default SinglePartner;
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/Partner/Partner.js:89]:
```suggestion
export default Partner;
```

**Review by kilemensi — APPROVED** [review]

**Pull request merged**